### ISSUE TITLE
Feature Request: Only publish `latest` for the default distribution channel

### ISSUE THREAD

**Issue by jftanner** [semantic-release-plus/semantic-release-plus#88]
Automatically publishing the `latest` tag is useful, but often undesirable for pre-releases, old-version fixes, etc. For example, if I'm publishing version `1.2.3-beta.1`, it's unlikely that I want it to also go out as `latest`.

Since there's no way to know in advance what the distribution channel will be, there's not a reasonable way to set `publishLatestTag` in the `release.config.json` file. Instead, `@semantic-release-plus/docker` would need to check the distribution channel and publish the tag _only_ if it's the default one.

I recognize that there may be some use-cases which need the `latest` tag in other circumstances, so perhaps it could be made configurable.

**Comment by JoA-MoS** [comment]
Ok, I understand... I will investigate looking that context to see when we can identify pre-release. I will likely leverage the same logic in something like semantic-release/npm.

**Comment by jftanner** [comment]
I did a little digging myself, and have opened a draft PR with a _really dirty and undesirable_ implementation. Feel free to take a look at it, or ignore it, as you choose.

**Comment by jftanner** [comment]
I've updated my PR (and taken it out of draft) with more sensible behavior based on the branch type rather than the distribution channel.

**Comment by JoA-MoS** [comment]
Ok, While reviewing this I recalled that at some point I was planning on adding prerelease and maintenance support. Similar to how the rest of semantic release does. When this plugin was originally written Semantic Release didn't have the pre-release functionality now that it does this plugin needs to be updated to support it the functionality that I would like to implement is the following

If the branch is a pre-release or maintenance branch we publish to a branch name tag.  In the case of docker a tagged item like  3.3.x, alpha, beta. is similar to an npm dist-tag. I want to simulate that same functionality.

So if using the docker plugin and merged to a pre-release branch name `alpha` you would end up with two tags pushed
```
alpha <-- this is always the latest of the alpha branch
1.2.3-alpha.1
```
The same would go for a maintenance branch if merged to a maintenance branch `5.2.x` you would have two tags pushed
```
5.2.x <-- this is always the latest on the maintenance branch
5.2.1
```
I think this covers the changes you request and will definitely be considering a breaking change.

**Comment by JoA-MoS** [comment]
Release with #108 on https://github.com/semantic-release-plus/semantic-release-plus/releases/tag/docker-v3.0.1
